src/object.c

```diff
@@ -168,7 +168,9 @@ robj *createStringObjectFromLongLongWithOptions(long long value, int valueobj) {
             o->encoding = OBJ_ENCODING_INT;
             o->ptr = (void*)((long)value);
         } else {
-            o = createObject(OBJ_STRING,sdsfromlonglong(value));
```

Comment:
I like this idea. Indeed many places can be replaced, but I found that sdsfromlonglong cannot be completely deleted, there are still places that are needed, for example https://github.com/redis/redis/blob/e4d183afd33e0b2e6e8d1c79a832f678a04a7886/src/geo.c#L292-L293

Another more general idea, not just for long long types, we can always be sure that some sds is temporary, small and must be freed before function ended, so we can safely use a static sds in the stack space to avoid memory alloc/free in the heap space. 
Not sure if it works, looking forward to some suggestions

Reply:
i think the speedup is not just because it avoids an extra allocation, i think it's also because the algorithm in ll2string is more optimal.
so in places that call `createObject(OBJ_STRING` (most notably in `rdbLoadIntegerObject`), we can completely replace the code to use `ll2string` and `createStringObject` (or maybe add a wrapper named ll2stringobj).
in other places that require an sds, we can keep using sdsfromlonglong, but delete the old implementation, and instead create one that uses ll2string internally.

Reply:
ooh, I get your ideas, thanks for your detailed description. Before I get started as your advises, I want to clarify a bit about the code: sds should not depend on the functions in util.h, so I want to copy the ll2string related functions from util.c into sds.c as an independent copy. But this seems a bit strange..

Reply:
you are right to be concerned, in the past sds.c was an independent library, but that's no longer the case, we consider it just part of redis nowadays (same as rax.c, dict.c, listpack.c, etc).
that said, it'll be slightly ugly to have it depend on utils.c

what i had in mind is to delete `sdsfromlonglong` from sds.c, and have an alternative implementation in utils.c
in theory, we could have renamed it to `ll2sds` or something, but then we'll have to touch many places in the code to update the new name, and i think that's unnecessary (unless after the other changes we end up having only a few of these)

Reply:
thank you, I prefer your second method. I love the sds.c for the pretty/simple and don't want don't want it to be ugly, at least not now and here :-)
I still keeped the name `sdsfromlonglong`, so that we would not to changes file every where.

Reply:
Infact the `sdsfromlonglong` functin was not exported to `sds.h` before then. This function is not declared in sds.h but in hiredis